Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -461,6 +466,66 @@ def add_comment(self, issue, text):
 
         return result
 
+    def issues_to_ids(self, issues):
+        if not isinstance(issues, list):
+            raise TypeError('Input should be a list of Issue objects.')
+        issue_ids = []
+        for i in issues:
+            try:
+                issue_id = i.link
+            except AttributeError:
+                raise AttributeError('Input should be a list of Issue objects.')
+            if isinstance(i.tracker, Tracker):
+                issue_ids.append(i.id)
+            else:
+                raise TypeError('Cannot relate issues of different types.')
+        return issue_ids
+
+    def relate(self, issue, depends_on=None, blocks=None, regressed_by=None, regressions=None, **relations):
+        if relations:
+            raise TypeError("'{}' is an invalid relation".format(list(relations.keys())[0]))
+
+        update_dict = dict()
+        update_dict['ids'] = [issue.id]
+        if depends_on:
+            update_dict['depends_on'] = {'add': self.issues_to_ids(depends_on)}
+        if blocks:
+            update_dict['blocks'] = {'add': self.issues_to_ids(blocks)}
+        if regressed_by:
+            update_dict['regressed_by'] = {'add': self.issues_to_ids(regressed_by)}
+        if regressions:
+            update_dict['regressions'] = {'add': self.issues_to_ids(regressions)}
+
+        response = None
+        try:
+            response = self.session.put(
+                '{}/rest/bug/{}{}'.format(self.url, issue.id, self._login_arguments(required=True)),
+                json=update_dict,
+                timeout=self.timeout,
+            )
+        except requests.exceptions.RequestException as e:
+            sys.stderr.write('Request Error: {}\n'.format(e))
+        except RuntimeError as e:
+            sys.stderr.write('Runtime Error:{}\n'.format(e))
```

Comment:
I don't think we should catch a runtime error here, I don't see how we would raise one.